### PR TITLE
kvdb: make kvdb a top level submodule to allow dependency in other projects

### PR DESCRIPTION
**Pull request by bhandras**
This PR makes `kvdb` a top level module in LND which will allow us to use it in other projects (eg. a quick boltbrowser fork: github.com/bhandras/boltbrowser).

Only the last 3 commits are part of this PR.

depends on: 
~- https://github.com/lightningnetwork/lnd/pull/5269~ (merged)
~- https://github.com/btcsuite/btcwallet/pull/745~ (merged)

### CODE REVIEW

File: kvdb/go.mod [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	github.com/tmc/grpc-websocket-proxy v0.0.0-20190109142713-0ad062ec5ee5 // indirect
+	go.etcd.io/bbolt v1.3.5-0.20200615073812-232d8fc87f50
+	go.etcd.io/etcd v3.4.14+incompatible
+	go.uber.org/zap v1.14.1 // indirect
+	golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899 // indirect
+)
+
+replace git.schwanenlied.me/yawning/bsaes.git => github.com/Yawning/bsaes v0.0.0-20180720073208-c0276d75487e
+
+replace github.com/lightningnetwork/lnd/healthcheck => github.com/bhandras/lnd/healthcheck v0.0.0-20210505133319-ad586c6bf1a2
```
Comment (guggero) [inline kvdb/go.mod:26]: Reminder to push a tag after merging #5269 and using that tag here instead.

File: kvdb/go.mod [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	github.com/lightningnetwork/lnd/healthcheck v1.0.0
+	github.com/stretchr/testify v1.7.0
+	github.com/tmc/grpc-websocket-proxy v0.0.0-20190109142713-0ad062ec5ee5 // indirect
+	go.etcd.io/bbolt v1.3.5-0.20200615073812-232d8fc87f50
+	go.etcd.io/etcd v3.4.14+incompatible
+	go.uber.org/zap v1.14.1 // indirect
+	golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899 // indirect
+)
+
+replace git.schwanenlied.me/yawning/bsaes.git => github.com/Yawning/bsaes v0.0.0-20180720073208-c0276d75487e
```
Comment (guggero) [inline kvdb/go.mod:24]: This replace isn't needed. We don't reference the `github.com/Yawning/aez` package here which has this indirect dependency.

**Review by guggero — APPROVED** [review]
LGTM barring dependent PRs #5269 and https://github.com/btcsuite/btcwallet/pull/745.

File: kvdb/go.mod (hunk 2 above)
Comment (bhandras) [reply kvdb/go.mod]: good catch, done!

File: kvdb/go.mod (hunk 1 above)
Comment (bhandras) [reply kvdb/go.mod]: done

**Review by Roasbeef — APPROVED** [review]
LGTM 🏓

**Pull request merged**